src/components/fields/TraceSelector.js

```diff
@@ -65,7 +65,13 @@ class TraceSelector extends Component {
       this.traceOptions = [{label: 'Scatter', value: 'scatter'}];
     }
 
-    this.fullValue = plotlyTraceToCustomTrace(props.fullContainer);
+    this.fullValue = plotlyTraceToCustomTrace(props.container);
+  }
+
+  componentWillMount() {
+    const {container} = this.props;
+    const value = plotlyTraceToCustomTrace(container);
+    this.updatePlot(value);
```

Comment:
we only want to do this if the mode isn't set though right? Also you are setting a custom trace type in user data which will break soon as they take their plot out of the editor app

Reply:
don't we want to aggressively set this in componentWillReceiveProps? What if the user at some random future time supplies `{x: [1], y: [1]}` in the gd.data?

Reply:
I'm calling updatePlot every time TraceSelector mounts, to make sure, that in any situation, the proper defaults are sent to plotly.js. 

So, in the case that a user starts out with x: [], y: [], type: 'scatter', before TraceSelector mounts, updatePlot will be called with {x: [], y: [], type: 'scatter', mode: 'markers'}, that way, on the next call to updatePlot, if the user provides new props of x: [1], y: [1], mode will already have been set to 'markers'

This is the reason I don't need to call this in componentWillReceiveProps, is this ok?

Reply:
It is usually better practice to only send an update _if_ you have to send an update but it probably won't be that costly so most likely won't make much of a difference. 

If you only call the function on component mount how will we aggressively repair a trace `{x: [1], y: [1]}` that comes into being on gd.data when this component is already mounted? If it is already mounted it won't send another update and this new trace will not be modified.

Reply:
> It is usually better practice to only send an update if you have to send
I DO have an update to send, I want to send the mode update, so that regardless of whether I have an empty state or a full one, the mode is set in gd.data.

So when x: [], y: [] changes to x:[1], y: [1] nothing additional needs to happen in componentWillReceiveProps because componentDidMount already set the mode in gd.data, and its gd.data that I use for trace recognition.

Reply:
I'll slack you. I don't think I am communicating this effectively